src/librustc_mir/interpret/memory.rs

```diff
@@ -73,26 +73,26 @@ impl MemoryPointer {
         MemoryPointer { alloc_id, offset }
     }
 
-    pub fn wrapping_signed_offset<'tcx>(self, i: i64, layout: &TargetDataLayout) -> Self {
-        MemoryPointer::new(self.alloc_id, value::wrapping_signed_offset(self.offset, i, layout))
+    pub(crate) fn wrapping_signed_offset<'a, L: HasDataLayout<'a>>(self, i: i64, l: L) -> Self {
```

Comment:
Ah, I didn't realize this trait already exists. :)

However, when I tried sth. similar, I came into trouble in the generic impl from `HasMemory` because of lifetimes not lasting long enough. The trait was slightly different though, maybe it works. (I will only be able to try this on Monday though.)